hermes/src/network/pythnet.rs

```diff
@@ -0,0 +1,173 @@
+            encoding: None,
+        })]),
+        with_context:   Some(true),
+    };
+
+    let (mut notif, _unsub) = client
+        .program_subscribe(&system_program::id(), Some(config))
+        .await?;
+
+    loop {
+        match notif.next().await {
+            Some(update) => {
+                let account: Account = match update.value.account.decode() {
+                    Some(account) => account,
+                    None => {
+                        log::error!("Failed to decode account from update: {:?}", update);
+                        continue;
+                    }
+                };
+
+                // The validators writes the accumulator messages using Borsh with
+                // the following struct. We cannot directly have messages as Vec<Messages>
+                // because they are serialized using big-endian byte order and Borsh
+                // uses little-endian byte order.
+                #[derive(BorshDeserialize)]
+                struct RawAccumulatorMessages {
+                    pub magic:        [u8; 4],
+                    pub slot:         u64,
+                    pub ring_size:    u32,
+                    pub raw_messages: Vec<Vec<u8>>,
+                }
+
+                let accumulator_messages = RawAccumulatorMessages::try_from_slice(&account.data);
+                match accumulator_messages {
+                    Ok(accumulator_messages) => {
+                        let messages = accumulator_messages
+                            .raw_messages
+                            .iter()
+                            .map(|message| {
+                                pythnet_sdk::wire::from_slice::<BE, Message>(message.as_slice())
+                            })
+                            .collect::<Result<Vec<Message>, _>>();
+
+                        let messages = match messages {
+                            Ok(messages) => messages,
+                            Err(err) => {
+                                log::error!("Failed to parse messages: {:?}", err);
+                                continue;
+                            }
+                        };
+
+                        let accumulator_messages = AccumulatorMessages {
+                            magic: accumulator_messages.magic,
+                            slot: accumulator_messages.slot,
+                            ring_size: accumulator_messages.ring_size,
+                            messages,
+                        };
+
+                        let (candidate, _) = Pubkey::find_program_address(
+                            &[
+                                b"AccumulatorState",
+                                &accumulator_messages.ring_index().to_be_bytes(),
+                            ],
+                            &system_program::id(),
+                        );
+
+                        if candidate.to_string() == update.value.pubkey {
```

Comment:
is there a reason this check is necessary, or is it just a sanity check? (maybe leave a comment)

Reply:
I guess there's a possibility that someone creates accounts that look like accumulator updates to confuse hermes?